app/frontend/src/features/profile/ProfileTagInput.tsx

```diff
@@ -0,0 +1,229 @@
+      borderWidth: 1,
+      borderStyle: "solid",
+      borderColor: "rgba(0, 0, 0, 0.23)",
+      boxShadow: theme.shadows[3],
+      zIndex: 1,
+      backgroundColor: theme.palette.background.default,
+      marginTop: theme.spacing(1),
+    },
+    button: {
+      borderRadius: theme.shape.borderRadius,
+      borderColor: "rgba(0, 0, 0, 0.23)",
+      borderWidth: 1,
+      borderStyle: "solid",
+      padding: "18.5px 14px",
+      margin: theme.spacing(1, 0),
+      width: "100%",
+      fontSize: "16px",
+      justifyContent: "space-between",
+    },
+    tag: {
+      padding: "0 14px",
+      margin: theme.spacing(1, 0),
+      fontSize: theme.typography.fontSize,
+      display: "flex",
+      alignItems: "center",
+    },
+    tagLabel: {
+      marginLeft: theme.spacing(1),
+    },
+    header: {
+      fontSize: theme.typography.fontSize,
+      borderBottomWidth: 1,
+      borderBottomStyle: "solid",
+      borderBottomColor: theme.palette.divider,
+      padding: theme.spacing(1, 2),
+    },
+    inputBase: {
+      padding: theme.spacing(2),
+      width: "100%",
+      borderBottomWidth: 1,
+      borderBottomStyle: "solid",
+      borderBottomColor: theme.palette.divider,
+      "& input": {
+        borderRadius: theme.shape.borderRadius,
+        backgroundColor: theme.palette.common.white,
+        padding: theme.spacing(1),
+        transition: theme.transitions.create(["border-color", "box-shadow"]),
+        borderWidth: 1,
+        borderStyle: "solid",
+        borderColor: theme.palette.divider,
+        "&:focus": {
+          boxShadow: `${fade(theme.palette.primary.main, 0.25)} 0 0 0 0.2rem`,
+          borderColor: theme.palette.primary.main,
+        },
+      },
+    },
+    paper: {
+      boxShadow: "none",
+      margin: 0,
+    },
+    option: {
+      minHeight: "auto",
+      alignItems: "flex-start",
+      padding: theme.spacing(1),
+      '&[aria-selected="true"]': {
+        backgroundColor: "transparent",
+      },
+      '&[data-focus="true"]': {
+        backgroundColor: theme.palette.action.hover,
+      },
+    },
+    popperDisablePortal: {
+      position: "relative",
+    },
+    checkbox: {
+      padding: 0,
+      marginRight: theme.spacing(1),
+    },
+  })
+);
+
+interface ProfileTagInputProps {
+  onChange: (_: any, value: string[]) => void;
+  value: string[];
+  options: string[];
+  label: string;
+  id: string;
+}
+
+export default function ProfileTagInput({
+  onChange,
+  value,
+  options,
+  label,
+  id,
+}: ProfileTagInputProps) {
+  const classes = useStyles();
+
+  const [open, setOpen] = useState<boolean>(false);
+  const anchorEl = useRef<null | HTMLButtonElement>(null);
+  const [valueState, setValueState] = useState<string[]>(value);
+  const [pendingValue, setPendingValue] = useState<string[]>([]);
+
+  useEffect(() => {
+    setValueState(value);
+  }, [value]);
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
```

Comment:
```suggestion
  const handleClick = () => {
```
Got a lint warning on this being unused